### DISCUSSION TITLE
Component script is very insistent on its name

### DISCUSSION THREAD

**Discussion by adpce** [GrapesJS/grapesjs#5383] [Q&A] [answered]
While adding a script to a custom component type, I follow the steps in the [documentation] (https://grapesjs.com/docs/modules/Components-js.html), and all works well.
However, I change the name of the script from `script` to `componentScript` or any other name, and then it breaks silently.
Is this intended behaviour, because it seems weird that the name of the script is fixed in such a way?

**Comment by artf** [comment]
That sounds similar to [this issue](https://github.com/GrapesJS/grapesjs/issues/5352) but not sure without a proper reproducible demo.

**Reply by adpce** [reply]
This issue doesn't produce broken javascript in final output, it produces no output at all. Dragging a component with the custom type into a canvas produces an empty div.

To reproduce, simply create a custom component type as outlined in the documentation, then change the script name to something other than 'script'.

Never used JSFiddle before, but this is my best attempt at a reproduction: https://jsfiddle.net/k02fuaz5/

**Reply by artf** [reply, marked as answer]
looks like you're confusing the variable name with the component property name, the property key has to be `script`, it can't be whatever you want...
```js
defaults: {
  script: firstscript
}
```

**Reply by adpce** [reply]
Thank you.
Please update docs to reflect this: https://grapesjs.com/docs/modules/Components-js.html
